tests/system/src/test/java/test/robot/javafx/scene/NodeInitializationBackgroundThreadTest.java

```diff
@@ -0,0 +1,906 @@
+            c.getData().setAll(createNumberSeries());
+            return c;
+        }, (c) -> {
+            c.getData().setAll(createNumberSeries());
+            accessChart(c);
+        });
+    }
+
+    @Disabled("JDK-8347392") // FIX
+    @Test
+    public void button() {
+        test(() -> {
+            Button c = new Button();
+            c.setSkin(new ButtonSkin(c));
+            return c;
+        }, (c) -> {
+            accessControl(c);
+            c.setAlignment(Pos.CENTER);
+            c.setText(nextString());
+            c.setDefaultButton(true);
+        });
+    }
+
+    @Test
+    public void canvas() {
+        test(() -> {
+            return new Canvas(30, 30);
+        }, (c) -> {
+            accessNode(c);
+            GraphicsContext g = c.getGraphicsContext2D();
+            g.setFill(Color.RED);
+            g.setStroke(Color.BLACK);
+            g.fillRect(5, 5, 5, 5);
+        });
+    }
+
+    @Disabled("JDK-8347392") // FIX
+    @Test
+    public void checkBox() {
+        test(() -> {
+            CheckBox c = new CheckBox("checkbox");
+            c.setSkin(new CheckBoxSkin(c));
+            return c;
+        }, (c) -> {
+            c.setAllowIndeterminate(true);
+            c.setSelected(true);
+            accessControl(c);
+        });
+    }
+
+    @Disabled("JDK-8347392") // FIX
+    @Test
+    public void choiceBox() {
+        test(() -> {
+            ChoiceBox c = new ChoiceBox();
+            c.setSkin(new ChoiceBoxSkin(c));
+            return c;
+        }, (c) -> {
+            c.getItems().setAll("ChoiceBox", "1", "2");
+            c.getSelectionModel().select(0);
+            accessControl(c);
+        });
+    }
+
+    @Test
+    public void colorPicker() {
+        test(() -> {
+            ColorPicker c = new ColorPicker();
+            c.setSkin(new ColorPickerSkin(c));
+            c.setValue(Color.GREEN);
+            return c;
+        }, (c) -> {
+            c.show(); // does not fail here, unlike DatePicker?
+            c.setValue(Color.RED);
+            c.prefHeight(-1);
+            c.setValue(Color.BLACK);
+            c.prefWidth(-1);
+            accessControl(c);
+        });
+    }
+
+    @Test
+    public void comboBox() {
+        test(() -> {
+            ComboBox c = new ComboBox();
+            c.setSkin(new ComboBoxListViewSkin(c));
+            c.getItems().setAll("ComboBox", "1", "2");
+            return c;
+        }, (c) -> {
+            c.setEditable(true);
+            c.getItems().setAll("ComboBox", nextString(), "2");
+            c.getSelectionModel().select(0);
+            accessControl(c);
+            c.show(); // does not fail here
+        });
+    }
+
+    @Disabled("JDK-8349004") // FIX
+    @Test
+    public void datePicker() {
+        test(() -> {
+            DatePicker c = new DatePicker();
+            c.setSkin(new DatePickerSkin(c));
+            return c;
+        }, (c) -> {
+            c.show(); // fails here
```

Comment:
As mentioned in the JBS issue, I think we should consider filing an issue to update the spec of `ComboBoxPopupControl::show` to require calling it on the JavaFX app thread (throwing an exception if called off thread). Either way, we should ensure that the skins themselves never call show unless the control is showing.

Reply:
yes, we'll deal with later.
I think there might be a solution which allows show() of an unconnected popup, and if that we'll use the ticket to add a javadoc comment and modify the test.

Reply:
Yes, whatever we end up doing will be done as a follow-up.